### PR TITLE
App template using Compose / package structure to support MVVM

### PR DESCRIPTION
**Pull request by Zermelo-101**

### CODE REVIEW

File: app/src/main/java/com/github/orkest/View/NavigationBarClasses.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import androidx.annotation.StringRes
+import com.github.orkest.R
+
+sealed class NavigationBarClasses(val route: String, @StringRes val resourceId: Int, val iconId : Int) {
+    object HomePage : NavigationBarClasses("HomePage", R.string.HomePage, R.drawable.icons8_audio_wave2_100)
+    object SearchPage : NavigationBarClasses("SearchPage", R.string.SearchPage,R.drawable.icons8_search)
+    object PlaylistPage : NavigationBarClasses("PlaylistPage",R.string.PlaylistPage,R.drawable.icons8_music_library_100)
+    object ProfilePage : NavigationBarClasses("ProfilePage",R.string.ProfilePage,R.drawable.icons8_test_account_100)
+    companion object {val  listOfNavigationItems = listOf(HomePage, SearchPage,PlaylistPage,ProfilePage)}
```
Comment (saraanej) [inline app/src/main/java/com/github/orkest/View/NavigationBarClasses.kt:11]: LGTM

File: app/src/main/java/com/github/orkest/View/NavigationBar.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import androidx.navigation.NavGraph.Companion.findStartDestination
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+
+class NavigationBar {
+
+    companion object{
+    @OptIn(ExperimentalMaterial3Api::class)
```
Comment (saraanej) [inline app/src/main/java/com/github/orkest/View/NavigationBar.kt:28]: An experimental API could be subject to changes in the future (breaking changes, bugs or other issues). Do the benefits of using this particular one overweight its downsides ?

File: app/src/main/java/com/github/orkest/View/NavigationBar.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                                    // avoid building up a large stack of destinations
+                                    // on the back stack as users select items
+                                    popUpTo(navController.graph.findStartDestination().id) {
+                                        saveState = true
+                                    }
+                                    // Avoid multiple copies of the same destination when
+                                    // reselecting the same item
+                                    launchSingleTop = true
+                                    // Restore state when reselecting a previously selected item
+                                    restoreState = true
```
Comment (saraanej) [inline app/src/main/java/com/github/orkest/View/NavigationBar.kt:53]: Comments are appreciated!

**Review by saraanej — APPROVED** [review]
Looks good to me

File: app/src/main/java/com/github/orkest/View/NavigationBar.kt (hunk 2 above)
Comment (Zermelo-101) [reply app/src/main/java/com/github/orkest/View/NavigationBar.kt:28]: I understand the concerns, TBH i just used the one they used in all the compose tutorial "Build Jetpack Compose UIs with Material Design 3 Components, the next evolution of Material Design."

**Pull request merged**